tests/integration/test_enrollment/test_agentd_enrollment.py

```diff
@@ -0,0 +1,84 @@
+'''
+brief: This module verifies the correct behavior of Wazuh Agentd during the enrollment under different configurations.
+copyright:
+    Copyright (C) 2015-2021, Wazuh Inc.
+    Created by Wazuh, Inc. <info@wazuh.com>.
+    This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+metadata:
+    component:
+        - Agent
+    modules:
+        - Agentd
+    daemons:
+        - Agentd
+    operating_system:
+        - Ubuntu
+        - CentOS
+        - Windows
+    tiers:
+        - 0
+    tags:
+        - Enrollment
+        - Agentd
+'''
+
+import pytest
+import ssl
+import yaml
+from wazuh_testing.tools.monitoring import ManInTheMiddle
+from wazuh_testing.tools.services import control_service
+from conftest import *
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.win32, pytest.mark.tier(level=0), pytest.mark.agent]
+
+configurations = load_wazuh_configurations(configurations_path, __name__)
+
+# Fixtures
+@pytest.fixture(scope="module", params=configurations)
+def get_configuration(request):
+    """Get configurations from the module"""
+    return request.param
+
+
+@pytest.mark.parametrize('test_case', [case for case in tests])
+def test_agentd_enrollment(set_test_case, configure_socket_listener, configure_environment, create_certificates,
+                           set_keys, set_pass, test_case: list):
```

Comment:
Same as before

Reply:
Same as before regarding the set of a global variable with the current test-case for sage into a callback method.